Core/include/Acts/Utilities/AlgebraHelpers.hpp

```diff
@@ -186,10 +188,21 @@ inline ActsMatrix<A::RowsAtCompileTime, B::ColsAtCompileTime> blockedMult(
 /// @return The theta value
 template <typename MatrixType, typename ResultType = MatrixType>
 std::optional<ResultType> safeInverse(const MatrixType& m) noexcept {
+  constexpr int rows = MatrixType::RowsAtCompileTime;
+  constexpr int cols = MatrixType::ColsAtCompileTime;
+
   ResultType result;
   bool invertible = false;
 
-  m.computeInverseWithCheck(result, invertible);
+  if constexpr (rows > 4 || cols > 4) {
+    Eigen::FullPivLU<MatrixType> mFullPivLU(m);
+    if (mFullPivLU.isInvertible()) {
```

Comment:
I wonder if this can still cause fpes

Reply:
Can we create a test-case that should create an FPE?

Reply:
I added some tests. I am not sure, if FPEs get thrown, but the function at least returns `false` in cases where we would expect FPEs.